src/main.py

```diff
@@ -0,0 +1,56 @@
+# OMIT !/usr/bin/env python3
+from pathlib import Path
+import click
+from io import StringIO
+
+from src.option_select import select
+from src.run_cocci import run_cocci
+
+@click.command
+@click.option(
+    "-i",
+    "--input",
+    type=Path,
+    required=True,
+    help="Directory containing your C source files.",
```

Comment:
We should also be able to process individual files. If that Path points to a file, then we'll analyze that one file, if it's a directory path, then we want to run the patches against the whole directory.